encoding_negotiator.js

```diff
@@ -0,0 +1,17 @@
+var zlib = require('zlib')
+var Negotiator = require('negotiator')
+
+/**
+ * @const
+ * whether current node version has brotli support
+ */
+var hasBrotliSupport = 'createBrotliCompress' in zlib
+
+function negotiateEncoding (req, encodings) {
+  var negotiator = new Negotiator(req)
+
+  return negotiator.encodings(encodings, hasBrotliSupport ? ['br'] : ['gzip'])[0]
```

Comment:
There's the `encoding` (singular) method that can be used. And I'd combine it with the existing `deflate` check as shown in https://github.com/expressjs/compression/pull/197.

Reply:
I like that option, i hadn't thought of it.